bindings/java/src/org/sleuthkit/datamodel/blackboardutils/WebBrowserArtifactsHelper.java

```diff
@@ -0,0 +1,538 @@
+	 * Adds a Web History artifact
+	 *
+	 * @param url          url visited, required
+	 * @param accessTime   last access time, may be 0 if not available
+	 * @param referrer     referrer, may be empty or null
+	 * @param title        website title, may be empty or null
+	 * @param programName, application recording the history, may be empty or
+	 *                     null
+	 *
+	 * @return artifact created
+	 */
+	public BlackboardArtifact addWebHistory(String url, long accessTime,
+			String referrer, String title, String programName) {
+		return addWebHistory(url, accessTime, referrer, title, programName,
+				Collections.<BlackboardAttribute>emptyList());
+	}
+
+	/**
+	 * Adds a Web History artifact
+	 *
+	 * @param url                 url visited, required
+	 * @param accessTime          last access time, may be 0 if not available
+	 * @param referrer            referrer, may be empty or null
+	 * @param title               website title, may be empty or null
+	 * @param programName,        application recording the history, may be
+	 *                            empty or null
+	 * @param otherAttributesList other attributes, may be an empty list
+	 *
+	 * @return artifact created
+	 */
+	public BlackboardArtifact addWebHistory(String url, long accessTime,
+			String referrer, String title, String programName,
+			Collection<BlackboardAttribute> otherAttributesList) {
+
+		BlackboardArtifact webHistoryArtifact = null;
+		try {
+			Collection<BlackboardAttribute> attributes = new ArrayList<>();
+			// Create artifact
+			webHistoryArtifact = getAbstractFile().newArtifact(BlackboardArtifact.ARTIFACT_TYPE.TSK_WEB_HISTORY);
+
+			// Add basic attributes 
+			attributes.add(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_URL, getModuleName(), url));
+
+			addAttributeIfNotZero(accessTime, BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DATETIME_ACCESSED, attributes);
+			addAttributeIfNotNull(title, BlackboardAttribute.ATTRIBUTE_TYPE.TSK_TITLE, attributes);
+			addAttributeIfNotNull(referrer, BlackboardAttribute.ATTRIBUTE_TYPE.TSK_REFERRER, attributes);
+
+			addAttributeIfNotNull(programName, BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PROG_NAME, attributes);
+			addAttributeIfNotNull(extractDomain(url), BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DOMAIN, attributes);
+
+			webHistoryArtifact.addAttributes(attributes);
+			webHistoryArtifact.addAttributes(otherAttributesList);
+
+			// post artifact 
+			getSleuthkitCase().getBlackboard().postArtifact(webHistoryArtifact, getModuleName());
+		} catch (TskCoreException ex) {
+			logger.log(Level.SEVERE, "Unable to add bookmark artifact", ex); //NON-NLS
+			return null;
+		} catch (Blackboard.BlackboardException ex) {
+			logger.log(Level.SEVERE, String.format("Unable to post artifact %s", ((webHistoryArtifact != null) ? webHistoryArtifact.getArtifactID() : "")), ex);  //NON-NLS
+		}
+
+		// return the artifact
+		return webHistoryArtifact;
+	}
+
+	// TBD: this is duplicated in Autopsy/NetworkUtils
```

Comment:
We can create a util in TSK and have the Autopsy util delegate to it.

Reply:
Should this util class go in the datamodel package? or the newly created datamodel.blackboardutils package? I'm leaning towards latter. 

Or we create a new package altogether.

Reply:
I'm not sure what vision @bcarrier has for the evolution of the TSK Java bindings. I could see creating a new package or putting it in the orignal datamodel package. I would not put it in datamodel.blackboardutils because I think it has greater scope than that.

For the sake of this PR, I think we should not hold things up waiting for Brian to communicate his wishes. For now, we can live with the code duplication in the non-public methods where this code currently resides.